src/transaction/Transaction.js

```diff
@@ -851,6 +851,86 @@ export default class Transaction extends Executable {
         return this;
     }
 
+    /**
+     * This method removes a signature from the transaction based on the public key or signature provided.
+     *
+     * @param {PublicKey} publicKey - The public key associated with the signature to remove.
+     * @param {Uint8Array} [signature] - The signature to remove. If omitted, all signatures for the given public key will be removed.
+     * @returns {Uint8Array[]} The removed signatures.
+     */
+    removeSignature(publicKey, signature) {
+        // If the transaction isn't frozen, freeze it.
+        if (!this.isFrozen()) {
```

Comment:
Do we need this? I mean if we signed we already froze the txn?

Reply:
In most cases, yes, but we also have this check in the addSignature() method since it interacts with states, and we need to ensure that it is frozen.